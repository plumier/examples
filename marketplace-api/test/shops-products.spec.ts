import supertest from "supertest"
import createApp from "../src/app"
import { closeConnection, createShop, ignore, userToken } from "./_helper"


afterEach(async () => {
    await closeConnection()
})

describe("Shop Item", () => {
    it("Should able adding item by staff", async () => {
        const app = await createApp({ mode: "production" })
        const { staffs, shop } = await createShop(app)
        await supertest(app.callback())
            .post(`/api/v1/shops/${shop.id}/products`)
            .send({ name: "MacBook Pro 16 2020", basePrice: 2000, price: 2500 })
            .set("Authorization", `Bearer ${staffs[0]?.token}`)
            .expect(200)
        const { body } = await supertest(app.callback())
            .get(`/api/v1/shops/${shop.id}/products`)
            .set("Authorization", `Bearer ${staffs[0]?.token}`)
            .expect(200)
        expect(body[0]).toMatchSnapshot(ignore)
    })
    it("Should hide basePrice for non shop staff", async () => {
        const app = await createApp({ mode: "production" })
        const { staffs, shop } = await createShop(app)
        await supertest(app.callback())
            .post(`/api/v1/shops/${shop.id}/products`)
            .send({ name: "MacBook Pro 16 2020", basePrice: 2000, price: 2500 })
            .set("Authorization", `Bearer ${staffs[0]?.token}`)
            .expect(200)
        const { body } = await supertest(app.callback())
            .get(`/api/v1/shops/${shop.id}/products`)
            .set("Authorization", `Bearer ${userToken}`)
            .expect(200)
        expect(body[0]).toMatchSnapshot(ignore)
    })
})