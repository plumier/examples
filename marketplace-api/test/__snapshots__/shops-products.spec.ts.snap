// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Item Should able adding item by staff 1`] = `
Object {
  "basePrice": 2000,
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "MacBook Pro 16 2020",
  "price": 2500,
  "updatedAt": Any<String>,
}
`;

exports[`Shop Item Should hide basePrice for non shop staff 1`] = `
Object {
  "createdAt": Any<String>,
  "deleted": false,
  "id": Any<Number>,
  "name": "MacBook Pro 16 2020",
  "price": 2500,
  "updatedAt": Any<String>,
}
`;
